src/lib.rs

```diff
@@ -110,11 +110,12 @@ mod macros;
 
 pub mod io;
 pub mod nr;
+pub mod debug;
 
 /// Performs a semihosting operation
 #[inline(always)]
 #[cfg(target_arch = "arm")]
-pub unsafe fn syscall<T>(mut nr: usize, arg: &T) -> usize {
+pub unsafe fn syscall<T: Sized>(mut nr: usize, arg: T) -> usize {
```

Comment:
I think this may be overloading this function too much. I think it would be best just to have syscall0, syscall1, etc. like the [sc crate](https://docs.rs/sc/0.2.0/sc/platform/index.html) has.